pyro/contrib/gp/models/vsgp.py

```diff
@@ -173,10 +173,10 @@ def forward(self, Xnew, full_cov=False):
         :rtype: tuple(torch.Tensor, torch.Tensor)
         """
         self._check_Xnew_shape(Xnew)
-        tmp_sample_latent = self._sample_latent
+        # avoid sampling the unnecessary latent u
         self._sample_latent = False
         Xu, u_loc, u_scale_tril = self.guide()
-        self._sample_latent = tmp_sample_latent
+        self._sample_latent = True
```

Comment:
ditto: can you confirm that `self._sample_latent` is always `True`, or otherwise use save-and-restore?